vllm/v1/worker/tpu_model_runner.py

```diff
@@ -0,0 +1,1323 @@
+
+        prefill_request_ids = []
+        prefill_prompt_lens = []
+        prefill_token_ids = []
+        prefill_position_ids = []
+        prefill_attn_metadata = []
+
+        # DECODES are the first num_decodes REQUESTS.
+        # PREFILLS are the next num_reqs - num_decodes REQUESTS.
+        num_reqs = self.input_batch.num_reqs
+        num_decodes = num_reqs - self.num_new_reqs
+        for idx in range(num_decodes, num_reqs):
+            prefill_request_ids.append(self.input_batch.req_ids[idx])
+
+            prompt_len = num_scheduled_tokens[idx]
+            prefill_prompt_lens.append(prompt_len)
+
+            # STATIC SHAPE: prefills are padded to the next power of 2.
+            padded_prompt_len = _get_padded_prefill_len(prompt_len)
+            assert padded_prompt_len <= self.max_model_len
+
+            # TOKEN_IDS.
+            token_ids = torch.from_numpy(self.input_batch.token_ids_cpu[
+                idx, :padded_prompt_len].reshape(1, -1))
+            prefill_token_ids.append(token_ids.to(self.device))
+
+            # POSITIONS.
+            positions = self.prefill_positions[:, :padded_prompt_len]
+            prefill_position_ids.append(positions.to(self.device))
+
+            # SLOT_MAPPING.
+            # The "slot" is the "physical index" of a token in the KV cache.
+            # Look up the block_idx in the block table (logical<>physical map)
+            # to compute this.
+            block_table_cpu_tensor = self.input_batch.block_table.get_cpu_tensor(
+            )
+            block_numbers = block_table_cpu_tensor[idx, positions //
+                                                   self.block_size].reshape(
+                                                       1, -1)
+            block_offsets = positions % self.block_size
+            slot_mapping = block_numbers * self.block_size + block_offsets
+            # Set an out of range value for the padding tokens so that they
+            # are ignored when inserting into the KV cache.
+            slot_mapping[:, prompt_len:] = _PAD_SLOT_ID
+            slot_mapping = slot_mapping.long()
+
+            prefill_attn_metadata.append(
+                PallasMetadata(
+                    num_prefills=1,
+                    num_prefill_tokens=padded_prompt_len,
+                    num_decode_tokens=0,
+                    slot_mapping=slot_mapping.to(self.device),
+                    multi_modal_placeholder_index_maps=None,
+                    block_tables=None,
+                    context_lens=None,
+                    effective_query_lens=None,
+                ))
+
+        return PrefillInputData(
+            request_ids=prefill_request_ids,
+            prompt_lens=prefill_prompt_lens,
+            token_ids=prefill_token_ids,
+            position_ids=prefill_position_ids,
+            attn_metadata=prefill_attn_metadata,
+        )
+
+    def _prepare_decode_inputs(self) -> DecodeInputData:
```

Comment:
again, do we really need `_prepare_decode_inputs` in V1 architecture?
(i'm assuming this can be handled with `_prepare_inputs` already.)